src/foam/nanos/http/ServiceWebAgent.java

```diff
@@ -86,7 +87,7 @@ public void execute(X x) {
         resp.setHeader("Access-Control-Allow-Origin", "*");
       } else if ( ! req.getHeader("Origin").equals("null") ){
         URL url = new URL(req.getHeader("Origin"));
-        if ( ((VirtualHostRoutingServlet) http.getServletMappings()[0].getServletObject()).getHostMapping().containsKey(url.getHost()) )
+        if ( Arrays.asList(http.getHostDomains()).contains(url.getHost()) )
```

Comment:
Better if HttpServer had a method that let you do something like: http.containsHostDomain(url.getHost()),
and then on the setHostDomains() preSet it sorted the values and then use java.util.Arrays.binarySort() instead of asList() and contains(). This would then be O(lgn) instead of O(n) and wouldn't generate any garbage, instead of O(n) garbage.

Reply:
Also ask Danny where he got with Set support.